src/components/panel/panel.spec.js

```diff
@@ -1201,6 +1238,34 @@ describe('$mdPanel', function() {
       myButton = angular.element(document.querySelector('button'));
     });
 
+    it('should call onRemoving if provided when removing the panel from the DOM with animation config provided', function() {
+      var onRemoving = jasmine.createSpy();
+      var deferred = $q.defer();
+      var resolvedPromise = deferred.promise;
+
+      var onRemovingAnimation = {
+        'animateClose' : function() {
+          return resolvedPromise;
+        }
+      };
+
+      var animationHookConfig = angular.extend(DEFAULT_CONFIG, {'onRemoving': onRemoving});
```

Comment:
Ahh yes, so the issue here is that the `_animateClose` function doesn't execute [this line](https://github.com/angular/material/blob/feat/panel/src/components/panel/panel.js#L1294) of code unless `animation` is defined on the `config` object. 

If I add this from the beginning, a bunch of `open` code gets called (wherever this `if (!animationConfig)` check gets called) and the animation I have to construct becomes a little complicated. 

If I add the animation to the `config` object after the open, I can still get the code path in the link I sent above to trigger, and I don't need to construct a huge `animation` object.

Thinking more about it though, this is hardly black box testing. So I am okay with constructing a proper `animation` object, I will leave it up to your discretion @KarenParker.